Xplat/src/main/java/vazkii/patchouli/common/util/ItemStackUtil.java

```diff
@@ -26,19 +30,22 @@
 
 import org.jetbrains.annotations.Nullable;
 
-import java.util.*;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
 
 public final class ItemStackUtil {
 	private static final Gson GSON = new GsonBuilder().create();
 
 	private ItemStackUtil() {}
 
-	public static Triple<ResourceLocation, Integer, CompoundTag> parseItemStackString(String res) {
-		String nbt = "";
-		int nbtStart = res.indexOf("{");
-		if (nbtStart > 0) {
-			nbt = res.substring(nbtStart).replaceAll("([^\\\\])'", "$1\"").replaceAll("\\\\'", "'");
-			res = res.substring(0, nbtStart);
+	public static Triple<ResourceLocation, Integer, DataComponentMap> parseItemStackString(String res) {
```

Comment:
Maybe it's time for a purpose built record here

Reply:
Probably, I didn't want to change anything without talking it over first